scripts/deploy.ts

```diff
@@ -18,17 +19,50 @@ async function deployGateway(): Promise<any> {
   
   console.log("✅ Deployed Gateway: ", tx.transactionHash);
 
+  if (network.config.chainId !== undefined) {
+    await updateConfigFile(network.config.chainId, contract.address);
+  }
+  console.log(`Proxy Contract Address: ${contract.address}`);
+
+  return tx;
+}
+
+async function deployGateway(): Promise<any> {
+  await confirmContinue({
+    contract: "Gateway",
+    network: network.name,
+    chainId: network.config.chainId,
+  });
+
+  const factory = await ethers.getContractFactory("Gateway");
+  const contract = await factory.deploy();
+
+  const tx = await contract.deployTransaction.wait();
+  
+  console.log("✅ Deployed Gateway: ", tx.transactionHash);
+
   // const implementationAddress = await contract.implementation();
+  await hre.run("verify:verify", {
+		address: contract.address,
+	});
+
+  if (network.config.chainId !== undefined) {
+    await updateConfigFile(network.config.chainId, contract.address);
+  }
 
   console.log(`Proxy Contract Address: ${contract.address}`);
-  // console.log("Implementation Contract Address:", implementationAddress);
 
   return tx;
 }
 
+
 async function main() {
-  // Deploy Gateway
-  await deployGateway();
+  const response = await waitForInput("\nDo you want to deploy a new Gateway proxy? y/N\n");
+  if (response !== "y") {
```

Comment:
`response.toLowerCase() !== 'y'`

Reply:
i don't think this counts sha, but will still update it.